core/api/src/main/java/com/alipay/sofa/rpc/context/RpcInvokeContext.java

```diff
@@ -246,17 +307,29 @@ public String removeRequestBaggage(String key) {
      * @return 全部响应透传数据
      */
     public Map<String, String> getAllRequestBaggage() {
-        return requestBaggage;
+        Lock lock = rwlock.readLock();
+        lock.lock();
+        try{
+            return requestBaggage;
```

Comment:
Here, after the `readLock` is locked, the original map is returned directly, and the protection of the `writeLock` is broken when the map is traversed by an external party.

Reply:
You are right. I transform all the maps in `RpcInvokeContext` to ConcurrentHashMap, so they will not throw `ConcurrentModificationException`.